File: FineraApp/backend/FineraAPI/Models/Category.cs
//Models/Category.cs

using System.ComponentModel.DataAnnotations;

namespace FineraAPI.Models
{
    public class Category
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(50)]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        public string Type { get; set; } = string.Empty; // Income or Expense
        
        public string Color { get; set; } = "#007bff";
        public string? Icon { get; set; }
        public int? UserId { get; set; }
        public bool IsDefault { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // Navigation properties
        public virtual User? User { get; set; }
        public virtual ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
        public virtual ICollection<Budget> Budgets { get; set; } = new List<Budget>();
    }
}

File: FineraApp/frontend/finera_flutter_app/lib/models/category.dart
//models/category.dart

class CategoryModel {
  final int? id;
  final String name;
  final String type; // "Income" or "Expense"
  final String color; // hex like #007bff
  final String? icon; // icon name string

  const CategoryModel({
    this.id,
    required this.name,
    required this.type,
    required this.color,
    this.icon,
  });

  factory CategoryModel.fromJson(Map<String, dynamic> json) {
    return CategoryModel(
      id: json['id'] as int?,
      name: json['name'] as String? ?? '',
      type: json['type'] as String? ?? '',
      color: json['color'] as String? ?? '#007bff',
      icon: json['icon'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'type': type,
      'color': color,
      'icon': icon,
    };
  }
}